react/components/ButtonPlain/index.tsx

```diff
@@ -1,97 +1,61 @@
-import React, { FunctionComponent, Fragment, useEffect, RefObject } from 'react'
-import PropTypes, { InferProps } from 'prop-types'
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, {
+  Fragment,
+  useEffect,
+  MutableRefObject,
+  ReactNode,
+  FC,
+  useRef,
+} from 'react'
+import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
 import Spinner from '../Spinner'
-import { withForwardedRef, refShape } from '../../modules/withForwardedRef'
+import { withForwardedRef } from '../../modules/withForwardedRef'
+import mergeRef from '../../utilities/mergeRef'
 
-const SIZE = {
-  SMALL: 'small',
-  REGULAR: 'regular',
-  LARGE: 'large',
-}
-
-const VARIATION = {
-  PLAIN: 'plain',
-  INVERTED: 'inverted',
-  DANGER: 'danger',
-}
-
-const loaderSize = {
-  [SIZE.SMALL]: 15,
-  [SIZE.REGULAR]: 20,
-  [SIZE.LARGE]: 25,
-}
+type MouseEventHandler = (e: React.MouseEvent<HTMLElement>) => void
```

Comment:
Why not use `React.MouseEventHandler`?

Reply:
I did not even know that these types existed. Thanks for that!